smarts/core/smarts.py

```diff
@@ -73,12 +73,21 @@ def __init__(
         timestep_sec=0.1,
         reset_agents_only=False,
         zoo_addrs=None,
+        renderer_optional=False,
     ):
         self._log = logging.getLogger(self.__class__.__name__)
         self._sim_id = Id.new("smarts")
         self._is_setup = False
         self._scenario: Scenario = None
-        self._renderer = Renderer(self._sim_id)
+        try:
+            self._renderer = Renderer(self._sim_id)
+        except Exception as e:
+            if not renderer_optional:
+                raise e
+            self._log.info(
+                f"Unable to create Renderer; continuing without one.\nException was:  {e}"
+            )
```

Comment:
I would be tempted to use a warning rather than info here although I see that the warning is instead given in the sensors. That may be OK.

Reply:
Yeah, I was torn about this.  I chose `info` because if they explicitly specified `renderer_optional=True`, they might not consider the absence of Panda3D as something to be worried about.